src/drivers/wayland/wayland.c

```diff
@@ -0,0 +1,2555 @@
+    struct lw_surface_object* decoration[LW_NUM_DECORATIONS];
+#endif
+
+    int width;
+    int height;
+
+    bool resize_pending;
+    int resize_width;
+    int resize_height;
+
+    bool flush_pending;
+    bool shall_close;
+    bool closed;
+    bool maximized;
+    bool fullscreen;
+    uint64_t frame_counter;
+    bool frame_done;
+
+    lv_display_render_mode_t render_mode;
+};
+
+/*********************************
+ *   STATIC VARIABLES and FUNTIONS
+ *********************************/
+
+static struct lw_application application;
+
+static const struct wl_callback_listener wl_surface_frame_listener;
+static bool resize_window(lw_window_t* window, int width, int height);
+static struct lw_surface_object* create_surface_obj(struct lw_application* app,
+    lw_window_t* window,
+    enum lw_object_type type,
+    struct lw_surface_object* parent);
+
+#if LV_WAYLAND_TICK_GET
+static uint32_t tick_get_cb(void)
+{
+    struct timespec t;
+    clock_gettime(CLOCK_MONOTONIC, &t);
+    uint64_t time_ms = t.tv_sec * 1000 + (t.tv_nsec / 1000000);
+    return time_ms;
+}
+#endif
+
+/**
+ * @brief The frame callback called when the compositor has finished rendering
+ * a frame.It increments the frame counter and sets up the callback
+ * for the next frame the frame counter is used to avoid needlessly
+ * committing frames too fast on a slow system
+ *
+ * NOTE: this function is invoked by the wayland-server library within the compositor
+ * the event is added to the queue, and then upon the next timer call it's
+ * called indirectly from handle_wayland_input (via wl_display_dispatch_queue)
+ * @param void data the user object defined that was tied to this event during
+ * the configuration of the callback
+ * @param struct wl_callback The callback that needs to be destroyed and re-created
+ * @param time Timestamp of the event (unused)
+ */
+static void frame_done_wl_cb(void* data, struct wl_callback* cb, uint32_t time)
+{
+    struct lw_surface_object* obj;
+    lw_window_t* window;
+
+    wl_callback_destroy(cb);
+
+    obj = (struct lw_surface_object*)data;
+    window = obj->window;
+    window->frame_counter++;
+
+    LV_LOG_TRACE("frame: %ld done, new frame: %ld",
+        window->frame_counter - 1, window->frame_counter);
+
+    window->frame_done = true;
+}
+
+static const struct wl_callback_listener wl_surface_frame_listener = {
+    .done = frame_done_wl_cb,
+};
+
+static void shm_format(void* data, struct wl_shm* wl_shm, uint32_t format)
+{
+    struct lw_application* app = data;
+
+    switch (format) {
+#if (LV_COLOR_DEPTH == 32)
+    case WL_SHM_FORMAT_ARGB8888:
+        app->format = format;
+        break;
+    case WL_SHM_FORMAT_XRGB8888:
+        if (app->format != WL_SHM_FORMAT_ARGB8888) {
+            app->format = format;
+        }
+        break;
+#elif (LV_COLOR_DEPTH == 16)
+    case WL_SHM_FORMAT_RGB565:
+        app->format = format;
+        break;
+#elif (LV_COLOR_DEPTH == 8)
+    case WL_SHM_FORMAT_RGB332:
```

Comment:
This was removed  because   lv_drivers used to support RGB332 when using a color depth of 8bits up to version release/v8.4. Since version 9.0 however, LV_COLOR_FORMAT_NATIVE is set to L8